include/engine/trip/trip_farthest_insertion.hpp

```diff
@@ -182,11 +184,15 @@ std::vector<NodeID> FarthestInsertionTrip(const NodeIDIterator &start,
         {
             for (auto y = start; y != end; ++y)
             {
+                // don't repeat coordinates
+                if (*x == *y)
+                    continue;
+
                 const auto xy_dist = dist_table(*x, *y);
                 // SCC decomposition done correctly?
                 BOOST_ASSERT(xy_dist != INVALID_EDGE_WEIGHT);
 
-                if (xy_dist > max_dist)
+                if (xy_dist >= max_dist)
```

Comment:
@MoKob can you quickly explain why this is needed? I see that a cucumber test is failing if this is not fixed but my head cannot get around the reason what's wrong here.